File: codes/cpp/chap02/answer_A07.cpp
#include <iostream>
using namespace std;

int N, L[100009], R[100009];
int D, B[100009];
int Answer[100009];

int main() {
	// 입력
	cin >> D >> N;
	for (int i = 1; i <= N; i++) cin >> L[i] >> R[i];

	// 전일 비에 더한다
	for (int i = 1; i <= N; i++) {
		B[L[i]] += 1;
		B[R[i] + 1] -= 1;
	}

	// 누적 합을 구한다 → 출력
	Answer[0] = 0;
	for (int d = 1; d <= D; d++) Answer[d] = Answer[d - 1] + B[d];
	for (int d = 1; d <= D; d++) cout << Answer[d] << endl;
	return 0;
}

File: codes/python/chap02/answer_A07.py
# 입력
D = int(input())
N = int(input())
L = [ None ] * N
R = [ None ] * N
for i in range(N):
	L[i], R[i] = map(int, input().split())

# 전일 비에 더한다
B = [ 0 ] * (D+2)
for i in range(N):
	B[L[i]] += 1
	B[R[i]+1] -= 1

# 누적 합을 구한다
Answer = [ None ] * (D+2)
Answer[0] = 0
for d in range(1, D+1):
	Answer[d] = Answer[d - 1] + B[d]

# 출력
for d in range(1, D+1):
	print(Answer[d])
